### ISSUE TITLE
IButton: Using role="button" in defiance of ARIA guidelines

### ISSUE THREAD

**Issue by pkej** [inkline/inkline#347]
* **What is the expected and current behavior?**
I expected <i-button> to have type="button", but it also has role="button" 

* **Steps to reproduce**
Create an app, in the template add:
```html
<i-button>test</i-button>
```
Start det server.
Use developer tools in your browser
Inspect element
Note that role="button" attribute exists in <button>

* **Please tell us about your environment:**
    - Version: >=3.2.2
    - Browser: [Firefox 111.0 ]
    - Language: [TypeScript]

* **Other information**
  [ARIA: button role](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/button_role) states:
  

> The button role identifies an element as a button to assistive technology such as screen readers. A button is a widget used to perform actions such as submitting a form, opening a dialog, canceling an action, or performing a command such as inserting a new record or displaying information. Adding role="button" tells assistive technology that the element is a button but provides no button functionality. Use <button> or <input> with type="button" instead.

**Comment by alexgrozav** [comment]
Hey! Thank you for reporting this. Any chance you could create a PR to the `inkline@3` branch?

**Comment by pkej** [comment]
I'll try to get around to it tomorrow

**Comment by pkej** [comment]
Hi, I've done the minimum change I could for this issue. However, I'm new to testing with frameworks, and  the Vitest snapshots of IButton, IDropdown and  INumberInput continue to have the 'role="button"' in their output, and I'm not sure how to create new snapshots that are correct.

The tests I created for the change I did is passing, but without the Assertions for the snapshots it's problematic.

Any help is appreciated. Should I create the PR for getting help? Meanwhile, I'm googling and youtubing it :)
